WDP181202-12

damianmichalski opened the pull request:
Problem: there is no style for mobile view.

Solution: added mobile view in styles

pawel-schmidt commented on src/sass/components/_header.scss:
```diff
@@ -224,3 +224,47 @@ header {
+      .text-right {
+        ul {
+          li {
+            margin-left: 0.9rem;
```
Just curious - why uneven margin-left according to margin-right?

xkxd commented on src/sass/components/_header.scss:
```diff
@@ -224,3 +224,47 @@ header {
+        flex-grow: 2;
+
+        ul {
+          li {
```
please reduce the nesting here.

this will result in `header .top-bar .text-left ul li` selector, which is hard to read and maintain. consider adding a class to `ul`/`li` and target them specifically via `.your-new-ul-class li` or ever `.your-new-li-class`

same applies to selector below

xkxd commented on src/sass/components/_header.scss:
```diff
@@ -224,3 +224,47 @@ header {
   }
 }
+@media (max-width: 767px) {
+  header {
```
do you have to nest every selector below in `header`?

damianmichalski commented:
@pawel-schmidt  uneven margin-left according to margin-right -> hmm i don't know :) i've set the same value. Also i replaced my @media query to boostrap built-in mixins
@xkxd i've added classes to `li` elements

xkxd commented on src/index.html:
```diff
@@ -16,16 +16,16 @@
-              <li><a href="#"><i class="fas fa-user"></i> Login</a></li>
-              <li><a href="#"><i class="fas fa-lock"></i> Register</a></li>
-              <li><a href="#"><i class="fas fa-bars"></i></a></li>
+              <li class="header-contact-li"><a href="#"><i class="fas fa-user"></i> <span class="header-links">Login</span></a></li>
```
what does `li` mean in that class name?

please don't use tag names in CSS classes, you'll have them separated better and you'll not run into situation when you'd like to use `header-contact-li` for other (eg. `div` element)

xkxd commented on src/sass/components/_header.scss:
```diff
@@ -224,3 +224,43 @@ header {
+      .text-left {
+        flex-grow: 2;
+
+        .header-first-li {
```
advice for the future - always think if it's necessary to have such long selectors as `header .top-bar .text-left .header-first-li` and try to reduce the nesting by applying styles either directly to specific classes (eg. `.header-first-li { ...`) or with minimal nesting possible eg. `.top-bar .header-first-li { ...`

according to SCSS guidelines 3-4 levels of nesting should be maximum.

damianmichalski commented:
@xkxd I've changed class names for `li` elements, please double check it.
And your advice is proper, but I  had to do it in this way to overwrite styles in the desktop version. In your approach i need also change style on desktop view.

xkxd commented:
please resolve conflicts
